<?xml version="1.0" ?>


<sdf version="1.5">
  <model name="train_model">
    <static>true</static>   <!-- train must be unmovable -->
    <link name="axis">
        <pose>0.0 1.5 2.0 0.0 1.5707963267948966 1.5707963267948966</pose>
        <collision name="collision">
            <geometry>
                <cylinder>
                    <radius>0.1</radius>
                    <length>3.0</length>
                </cylinder>
            </geometry>
        </collision>
        <visual name="visual">
            <geometry>
                <cylinder>
                    <radius>0.1</radius>
                    <length>3.0</length>
                </cylinder>
            </geometry>
        </visual>
    </link>
    <link name="left_wheel">
        <pose>0.0 0.0 2.0 0.0 1.5707963267948966 1.5707963267948966</pose>
        <collision name="collision">
            <geometry>
                <cylinder>
                    <radius>1.0</radius>
                    <length>0.4</length>
                </cylinder>
            </geometry>
        </collision>
        <visual name="visual">
            <geometry>
                <cylinder>
                    <radius>1.0</radius>
                    <length>0.4</length>
                </cylinder>
            </geometry>
        </visual>
    </link>
    <link name="right_wheel">
        <pose>0.0 3.0 2.0 0.0 1.5707963267948966 1.5707963267948966</pose>
        <collision name="collision">
            <geometry>
                <cylinder>
                    <radius>1.0</radius>
                    <length>0.4</length>
                </cylinder>
            </geometry>
        </collision>
        <visual name="visual">
            <geometry>
                <cylinder>
                    <radius>1.0</radius>
                    <length>0.4</length>
                </cylinder>
            </geometry>
        </visual>
    </link>
    <link name="left_disk">
        <pose>0.0 1.0 2.0 0.0 1.5707963267948966 1.5707963267948966</pose>
        <collision name="collision">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.08</length>
                </cylinder>
            </geometry>
        </collision>
        <visual name="visual">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.08</length>
                </cylinder>
            </geometry>
        </visual>
    </link>
    <link name="right_disk">
        <pose>0.0 2.0 2.0 0.0 1.5707963267948966 1.5707963267948966</pose>
        <collision name="collision">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.08</length>
                </cylinder>
            </geometry>
        </collision>
        <visual name="visual">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.08</length>
                </cylinder>
            </geometry>
        </visual>
    </link>
    <link name="left_pad">
        <pose>0.0 1.09 2.0 0.0 1.5707963267948966 1.5707963267948966</pose>
        <collision name="collision">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.1</length>
                </cylinder>
            </geometry>
        </collision>
        <visual name="visual">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.1</length>
                </cylinder>
            </geometry>
            <material>
                <ambient>1.0 0.0 0.0 1</ambient>
            </material>
        </visual>
    </link>
    <link name="right_pad">
        <pose>0.0 1.91 2.0 0.0 1.5707963267948966 1.5707963267948966</pose>
        <collision name="collision">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.1</length>
                </cylinder>
            </geometry>
        </collision>
        <visual name="visual">
            <geometry>
                <cylinder>
                    <radius>0.25</radius>
                    <length>0.1</length>
                </cylinder>
            </geometry>
            <material>
                <ambient>1.0 0.0 0.0 1</ambient>
            </material>
        </visual>
    </link>
  </model>
</sdf>
